.github/workflows/validate_company_industry.yml

```diff
@@ -2,15 +2,16 @@ name: Validate Company Industry
 
 on:
   pull_request:
-    paths:
-      - "priv/companies/*.exs" # only run on changes to company files
 
 jobs:
   validate_struct:
     runs-on: ubuntu-latest
 
     steps:
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
```

Comment:
Use this action instead. It can return a list of changed company files and handles extending the depth as needed. That way we can still have fast clones.

https://github.com/tj-actions/changed-files

Reply:
Dang you are so wise @btkostner 🙇 . I looked up `get changed files github actions` and got https://github.com/jitterbit/get-changed-files which has been out of date for 3+ years so I thought to go with some bash-magic I should have searched around a bit more. 

But putting a positive spin on this, I learned a bunch about bash and github context in workflows.

Reply:
updated! ty for the grand suggestion again.